save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TestRepository.kt

```diff
@@ -3,14 +3,15 @@ package org.cqfn.save.backend.repository
 import org.cqfn.save.entities.Test
 import org.springframework.stereotype.Repository
 
+import java.util.Optional
+
 /**
  * Repository of tests
  */
 @Repository
+@Suppress("MISSING_KDOC_CLASS_ELEMENTS", "MISSING_KDOC_ON_FUNCTION")
 interface TestRepository : BaseEntityRepository<Test> {
-    /**
-     * @param hash
-     * @return Test was founded by hash
-     */
-    fun findByHash(hash: String): Test?
+    fun findByHashAndFilePath(hash: String, filePath: String): Optional<Test>
+
+    fun findByHashAndFilePathAndTestSuiteId(hash: String, filePath: String, testSuiteId: Long): Optional<Test>
```

Comment:
Is `TestSuiteId` necessary here?

Reply:
Why not just `Test?` instead of `Optional`

Reply:
Because the same file can be a test in different test suites. Also, using `find(...).orElseGet { ... }.id` seemed nicer than `(find(...) ?: run { ... }).id`